src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -105,6 +105,14 @@ function checkAndWarnForMutatedStyle(style1, style2, component) {
   if (shallowEqual(style1, style2)) {
     return;
   }
+  // Don't show mutated style warning if a value is NaN since the styles weren't mutated.
+  for (var key in style1) {
+    var value = style1[key];
+
+    if (isNaN(value) && ('' + value) === 'NaN') {
+      return;
```

Comment:
I originally thought the NaN warning would suffice, but you are right and it would be a good idea to show both. I've updated the PR to remove the NaN values before the equality check for the mutation warning.